### PR TITLE
Fix responsePrefix correctness on failover + prevent duplicate prefixes

### PR DESCRIPTION
**Pull request by Monkey-D-Luisi**
## Summary

- **Duplicate prefix after failover (all dispatcher channels)**: `normalizeReplyPayload` prepend-only guard (`!text.startsWith(effectivePrefix)`) failed when the model changed between block and final replies, causing `[old] [new] msg`. Fixed by tracking `lastAppliedPrefix` in the dispatcher and stripping the old prefix before applying the new one, with whitespace boundary check to prevent false substring matches.
- **Webchat streaming deltas missing prefix**: `emitChatDelta` broadcast raw buffer text without prefix; only `emitChatFinal` applied it. Fixed by prepending the resolved prefix in delta broadcasts (buffer stays raw for the definitive final).

## Test plan

- [x] 5 unit tests in `reply-utils.test.ts` (strip-then-replace, no-strip-when-undefined, idempotency, whitespace-terminated prefix, substring edge case)
- [ ] Telegram `streamMode: "partial"` with forced failover: verify single prefix `[fallback]` instead of `[primary] [fallback]`
- [ ] Webchat with forced failover: verify streaming deltas and final both show fallback model prefix
- [ ] No-failover regression: verify prefix behavior unchanged when primary model succeeds

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/auto-reply/reply/normalize-reply.ts [hunk 1]
```diff
@@ -81,13 +84,20 @@ export function normalizeReplyPayload(
...
+  if (effectivePrefix && text && text.trim() !== HEARTBEAT_TOKEN) {
+    // If a previous (different) prefix was applied, strip it before applying the current one.
+    // This prevents double-prefix after model fallback (e.g. block reply with model A, final with model B).
+    if (
+      opts.previouslyAppliedPrefix &&
+      opts.previouslyAppliedPrefix !== effectivePrefix &&
+      text.startsWith(opts.previouslyAppliedPrefix)
+    ) {
+      text = text.slice(opts.previouslyAppliedPrefix.length).trimStart();
+    }
```
[the remark this answers is not in the document]
Comment (Monkey-D-Luisi) [reply src/auto-reply/reply/normalize-reply.ts:101]: Fixed in 1099291. Added boundary check: strip only proceeds if the character after the old prefix is whitespace (our separator) or the prefix itself ends with whitespace (e.g. `[model]\n`). This prevents `[gpt]` from matching `[gpt-4]` since `-` is not whitespace.

File: src/auto-reply/reply/reply-utils.test.ts [hunk 2]
```diff
@@ -114,6 +114,60 @@ describe("normalizeReplyPayload", () => {
...
+      { text: "pong" },
+      {
+        responsePrefix: "[{provider}/{model}]\n",
+        responsePrefixContext: { provider: "anthropic", model: "claude-sonnet" },
+      },
+    );
+
+    expect(normalized).not.toBeNull();
+    expect(normalized?.text).toBe("[anthropic/claude-sonnet]\npong");
+  });
```
[the remark this answers is not in the document]
Comment (Monkey-D-Luisi) [reply src/auto-reply/reply/reply-utils.test.ts:170]: Added in 1099291. New test: `"does not strip when old prefix is a substring of text prefix"` — verifies `previouslyAppliedPrefix: "[gpt]"` does not corrupt `"[gpt-4] hello"`.

File: src/auto-reply/reply/reply-dispatcher.ts [hunk 3]
```diff
@@ -123,16 +132,31 @@ export function createReplyDispatcher(options: ReplyDispatcherOptions): ReplyDis
...
+        lastAppliedPrefix !== currentResolvedPrefix ? lastAppliedPrefix : undefined,
     });
     if (!normalized) {
       return false;
     }
+
+    // Track the prefix that was just applied.
+    if (currentResolvedPrefix) {
+      lastAppliedPrefix = currentResolvedPrefix;
+    }
```
[the remark this answers is not in the document]
Comment (Monkey-D-Luisi) [reply src/auto-reply/reply/reply-dispatcher.ts]: Fixed in 1099291. `lastAppliedPrefix` now updates unconditionally: `lastAppliedPrefix = currentResolvedPrefix;`

File: src/auto-reply/reply/agent-runner-execution.ts [hunk 4]
```diff
@@ -415,6 +415,18 @@ export async function runAgentTurnWithFallback(params: {
...
+
+      // Safety net: re-notify with the actual final provider/model after fallback.
+      // This is a no-op when no failover occurred (same values as the last run callback),
+      // but ensures the context is correct if fallback changed the model.
+      if (fallbackProvider && fallbackModel) {
+        params.opts?.onModelSelected?.({
+          provider: fallbackProvider,
+          model: fallbackModel,
+          thinkLevel: params.followupRun.run.thinkLevel,
+        });
```
[the remark this answers is not in the document]
Comment (Monkey-D-Luisi) [reply src/auto-reply/reply/agent-runner-execution.ts]: Agreed — removed in 1099291. The lifecycle `end` event fires before `runWithModelFallback` returns, so `emitChatFinal` cleans up `resolvedPrefixes` before the safety net runs, causing stale entries. The `onModelSelected` call inside each `run()` callback already ensures the last call has the correct final model, making the safety net redundant.

**Comment by Monkey-D-Luisi** [comment]
The `checks-windows (node, test)` failure is a pre-existing flaky test unrelated to this PR:

```
FAIL extensions/msteams/src/messenger.test.ts > msteams messenger > sendMSTeamsMessages > preserves parsed mentions when appending OneDrive fallback file links
AssertionError: expected "vi.fn()" to be called once, but got 0 times
```

This same test also fails on `main` — the CI workflow on the main branch has been consistently failing with the same error. None of the files changed in this PR touch MS Teams or OneDrive code.
